File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) {
        int x = n;
        int digitSum = 0;
        int prod = 1;
        while (x > 0){
            int digit = x % 10;
            digitSum += digit;
            prod *= digit;
            x /= 10;
        }
        return n % (digitSum + prod) == 0;
    }
}

File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        x = n
        digitSum = 0
        prod = 1
        while x > 0:
            digit = x % 10
            digitSum += digit
            prod *= digit
            x //= 10
        return n % (digitSum + prod) == 0